src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -249,16 +249,48 @@ async function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourc
   const existingPostRun = moduleConfig.postRun || [];
   (moduleConfig as any).preloadPlugins = [];
 
+  var totalResources = Object.keys(resources.assembly).length + Object.keys(resources.pdb || {}).length + /* dotnet.wasm */ 1;
```

Comment:
Nit: `const` instead of `var` here.

Though it might be slightly easier still if you were to concatenate all the promises into a single array without counting them, and then you can just do `thatArray.length` whenever we need to know the total number of resources.